Library/OcCryptoLib/Sha2.c

```diff
@@ -599,6 +560,11 @@ Sha384 (
 {
   SHA384_CONTEXT Ctx;
 
+  if (IsAvxSupported ()) {
```

Comment:
I would suggest not making a detection check for AVX implicitly. While many firmwares will not have any issues, I would still try to avoid doing it without explicit user consent. For this reason I suggest the following:

- Make a `TryEnableAvx` function (ex `IsAvxSupported`) accessible from `OcCryptoLib.h`
- Introduce a `extern BOOLEAN mIsAvxEnabled` inside Sha2Internals.h, which gets assigned by this function.
- In all the Sha2 functions check the boolean as needed instead of trying to auto-enable AVX.

In OC codebase we will call the function ourselves later as it will probably need some configuration thought. For Amaranth just go with   AmStorageLib probably?

CC @mhaeuser